### PR TITLE
add conditions in the function of perpendicular_momentum

### PR DESCRIPTION
**Pull request by Jiemin-Li**
Add some conditions at the beginning of the perpendicular_momentum such that (1) the input arguments must have same lengths; (2) or there is only one input argument whose length is larger than 1. @awalter-bnl Please have a look.

### CODE REVIEW

File: src/test_data/arpes.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import math
+from tabnanny import check
```
Comment (awalter-bnl) [inline src/test_data/arpes.py]: This input here appears not to be used, can we remove it? If I missed where it is used it is common to have imports in alphabetical order so this should be  after the scipy import.

File: src/test_data/arpes.py [hunk 2]
```diff
@@ -367,6 +369,32 @@ def perpendicular_momentum(photon_energy, parallel_momentum,
...
+    if not np.all(check_data_type):
+        for i in [photon_energy, parallel_momentum, binding_energy]:
+            if isinstance(i,(np.ndarray)):
+                if len(np.shape(i))>1:
+                    raise ValueError(f'The shape of {i} can not be '
+                                     f'larger than 1!')
+                else:
+                    if len(i)>1:
+                        check_data_len.append(len(i))
+            elif isinstance(i,(list)):
```
Comment (awalter-bnl) [inline src/test_data/arpes.py]: This `elif` is not required, as inputting these as lists will not work. The reason for this is the code below uses numpy arithmetic functions which are only guaranteed to work for numpy arrays, floats or ints. If you would like lists to work add some code before this that checks for lists in these inputs and convert them to numpy arrays.

**Review by awalter-bnl — COMMENTED** [review]
Perhaps this could be written a little more succinctly written as below. I think this will run faster than the multiple if statements that you are suggesting. But I let you decide, I have made a few other comments if you decide not to use what is below.

```
check_data = [len(value)
      if (isinstance(value,(np.ndarray)) and len(value.shape)==1)
      else False
      for value in [inner_value for inner_value in 
                    [parallel_momentum, photon_energy, 
                     binding_energy]
                    if not isinstance(inner_value, (float, int))]]

if not np.all(check_data_len):
    raise ValueError (f'Some string about not being an int, float or 1D numpy array')
elif check_data_len[:-1] != check_data_len[1:]:
    raise ValueError (f'Some string about all numpy arrays not being the same length')
```

File: src/test_data/arpes.py (hunk 2 above)
Comment (Jiemin-Li) [reply src/test_data/arpes.py]: Replace these lines with the new codes.

**Comment by awalter-bnl** [comment]
Looks good @Jiemin-Li , thanks for the PR

**Pull request merged**